src/Common/src/TypeSystem/Common/TypeSystemContext.cs

```diff
@@ -492,9 +492,6 @@ protected override MethodForInstantiatedType CreateValueFromKey(MethodForInstant
 
         public MethodDesc GetMethodForInstantiatedType(MethodDesc typicalMethodDef, InstantiatedType instantiatedType)
         {
-            Debug.Assert(!(typicalMethodDef is MethodForInstantiatedType));
-            Debug.Assert(!(typicalMethodDef is InstantiatedMethod));
-
             return _methodForInstantiatedTypes.GetOrCreateValue(new MethodForInstantiatedTypeKey(typicalMethodDef, instantiatedType));
```

Comment:
Why remove these asserts? What kkind of methods were appearing here as typicalMethodDefs?